src/fshtypes/CodeSystem.ts

```diff
@@ -0,0 +1,23 @@
+import { FshEntity } from './FshEntity';
+import { FshCode } from './FshCode';
+
+/**
+ * For more information about a CodeSystem in FHIR,
+ * @see {@link http://hl7.org/fhir/codesystem-definitions.html}
+ */
+export class CodeSystem extends FshEntity {
+  id: string;
+  title?: string;
+  description?: string;
+  concepts: FshCode[];
+
+  constructor(public name: string) {
+    super();
+    this.id = name;
+    this.concepts = [];
+  }
+
+  addConcept(code: string, display?: string) {
+    this.concepts.push(new FshCode(code, this.id, display));
```

Comment:
So... this is interesting, because really, the system in `FshCode` *ought* to be a URL -- but since we haven't yet generated the URL (it's based on the config `canonical`), we don't have a URL to put into the `FshCode`.

While I totally understand the preference for re-using `FshCode`, there is an added problem since CodeSystem concepts can have both a `display` _and_ a `definition` -- and `FshCode` has nowhere to put `definition`.

So... I wonder if it's better to create a new special type (or class) representing a CodeSystem concept, that has `code`, `display`, and `definition`, but does _not_ have `system`?

Reply:
A new class seems like the best approach, since as you pointed out, this really isn't quite the same as a `FshCode`. I've updated the branch accordingly.